modules/mono/utils/thread_local.h

```diff
@@ -118,11 +118,11 @@ class ThreadLocal {
 		_initialize(T());
 	}
 
-	ThreadLocal(const T &p_init_val) {
+	explicit ThreadLocal(const T &p_init_val) {
 		_initialize(p_init_val);
 	}
 
-	ThreadLocal(const ThreadLocal &other) {
+	ThreadLocal(const ThreadLocal &other) { // TODO explicit?
```

Comment:
Neither of these two constructors should be explicit.

Reply:
just double checking as cppcheck didnt pick them up, definitely not a perfect program, but i am using an older version annoyingly, will fix

Reply:
We have the macro `_THREAD_LOCAL_` that can be used as:

```C++
_THREAD_LOCAL_(int) foo = 20;

// If thread_local is supported, expands to...
thread_local int foo = 20;

// If neither thread_local, __thread or __declspec((thread) are supported
ThreadLocal<int> foo = 20; // needs implicit ThreadLocal(const T &p_init_val)
```